docs/20_smart-contracts/40_guides/10_create_smart_contracts_guide.md

```diff
@@ -0,0 +1,1351 @@
+
+Multi-index tables are often used in EOS smart contract development to store data that needs to be accessed frequently and efficiently, such as user account balances, transaction history, or other important information. Because multi-index tables are implemented as an on-chain database, they can be accessed and modified by other smart contracts, as well as by external applications through the EOSIO API.
+
+Here's an example of a simple multi-index table definition:
+
+```cpp
+TABLE user_balance {
+  name user;
+  asset balance;
+
+  uint64_t primary_key() const { return user.value; }
+};
+using balance_table = eosio::multi_index<"balances"_n, user_balance>;
+```
+
+The name of a multi-index table has the same restrictions as the name of an action.
+
+The code above defines a multi-index table type called `balance_table`. The table stores user account balances which are encapsulated in the `user_balance` structure. The structure contains two fields: the `name` field represents the user's account name, and the `asset` field represents the user's account balance. The `primary_key()` inline method implementation defines the primary key for the table, which in this case is the user's account name represented as a 64-bit unsigned integer value.
+
+Developers can use the `balance_table` template type, to instantiate a reference to a table and to perform various operations on that table, such as:
+
+- query the table for specific data,
+- insert new rows,
+- modify existing rows,
+- delete existing rows.
+
+### Multi-index: Code, Scope and Table
+
+When you define and use a multi-index table in an EOS smart contract, there are three important components to consider:
+
+- the table,
+- the code,
+- the scope.
+
+#### The table
+
+The table is the name of the multi-index table itself. This name is used to identify the table when performing operations on it, such as inserting, modifying, deleting, or querying data. The table name must be specified as a template parameter to the multi_index class.
+
+```cpp
+   using balance_table = eosio::multi_index<"balances"_n, user_balance>;
+```
+
+The above code defines the type `balance_table` as a multi-index table which has the name `balances` and is capable to store instance of `user_balance` types.
+
+#### The code
+
+The code is the account that owns the smart contract. This account is responsible for paying the RAM storage costs associated with the multi-index table. When defining a multi-index table, you must specify the code account as the first argument to the multi_index constructor.
+
+#### The scope
+
+The scope is a secondary identifier that is used to group related data within the multi-index table. When defining a multi-index table, you must specify the scope as the second argument to the multi_index constructor. The scope is often set to the contract account itself, to group all related data within the same contract.
+
+```cpp
+balance_table balances(get_self(), get_self().value);
```

Comment:
Can probably just this section down a lot with a diagram/text that just shows

```
table_type table_name(CODE, SCOPE);
```

Reply:
I think this will butcher the whole explanation and not teach the reader almost anything about the code and scope.

Reply:
Alright I'll give it a reread on Monday